go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -253,3 +255,15 @@ func egressGetL4Match(ports []egressfirewallapi.EgressFirewallPort) string {
 	}
 	return l4Match
 }
+
+func getClusterSubnetsExclusion() string {
+	var exclusion string
+	for _, clusterSubnet := range config.Default.ClusterSubnets {
+		if utilnet.IsIPv6CIDR(clusterSubnet.CIDR) {
+			exclusion += fmt.Sprintf(" || %s.dst != %s", "ip6", clusterSubnet.CIDR)
```

Comment:
this doesn't seem right to me. If the dest doesnt match any network, you have the value becoming true. For example, if my cluster subnets are A,B,C and the dest on my packet is A:
A || B || C would be: false || true|| true, which would result in true

That would then be && with your original match:
match = fmt.Sprintf("%s && (%s)\"", match, getClusterSubnetsExclusion())

and enforce your rule right? even though it should have been skipped cause its cluster subnet dest?

Reply:
it is && with the original match I am || if if there are multiple subnets togather so it will generate matches like below (taken from the testing)
 `ovn-nbctl --timeout=15 --id=@logical_router_policy create logical_router_policy priority=10000 match=\"ip4.dst == 1.2.3.4/23 && ip4.src == $a10481622940199974102 && (ip4.dst != 10.128.0.0/14)\" action=allow external-ids:egressFirewall=namespace1 -- add logical_router ovn_cluster_router policies @logical_router_policy"`

or in the case of mutiple cluster cidrs 
`ovn-nbctl --timeout=15 --id=@logical_router_policy create logical_router_policy priority=10000 match=\"ip4.dst == 1.2.3.4/23 && ip4.src == $a10481622940199974102 && (ip4.dst != 10.128.0.0/14 || ip4.dst != 10.0.0.0/14)\" action=allow external-ids:egressFirewall=namespace1 -- add logical_router ovn_cluster_router policies @logical_router_policy"`

am I thinking incorrectly?

Reply:
done, just AND'ed all the fields together